src/metabase/driver/util.clj

```diff
@@ -129,6 +129,21 @@
         In case you're connecting via an SSH tunnel and run into a timeout, you might consider increasing this value
         as the connections via tunnels have more overhead than connections without.")
 
+;; This is normally set via the env var `MB_DB_QUERY_TIMEOUT_MINUTES`
+(defsetting db-query-timeout-minutes
+  "By default, this is 20 minutes."
+  :visibility :internal
+  :export?    false
+  :type       :integer
+  ;; I don't know if these numbers make sense, but my thinking is we want to enable (somewhat) long-running queries on
+  ;; prod but for test and dev purposes we want to fail faster because it usually means I broke something in the QP
+  ;; code
+  :default    (if config/is-prod?
+                20
+                3)
+  :doc "Timeout in minutes for query execution, both Metabase application database and data connections.
+In case you're execute a query and run into a timeout, you might consider increasing this value.")
```

Comment:
We need a disclaimer here that this impacts only the Metabase backend. Really long queries (>20 minutes) will likely be dropped by other pieces like nginx or the database itself, and we can't guarantee this will be respected.